### PR TITLE
chore: specify TS 5.x support

### LINKED ISSUE #26204: Component Testing: `typescript ^=5.0.0` support
### Description

When I setup component testing typescript 5 is not detected as typescript.

### URL of Issue(s)

https://docs.cypress.io/guides/component-testing/vue/quickstart

### Steps to replicate

```shell
npm i typescript -D
npx cypress open
```

### Browser

_No response_

### Device

- [X] PC
- [ ] Mac
- [ ] iPhone
- [ ] iPad
- [ ] Android Phone
- [ ] Android Tablet

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes #26204

### Additional details

Specify Cypress CT works with TypeScript 5.x.

### Steps to test

You could make a project using CT and TS 5.x There's zillions of permutations (Next, Vue, Angular...). No doubt some specific combinations might not work, but in general, none of the [breaking changes](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/#breaking-changes-and-deprecations) impact us from what I can see.

### How has the user experience changed?

Now when setting up CT, users with TS 5.x installed won't feel like it's not supported when going through the onboarding workflow, which is currently the case: #26204. This makes for a bad onboarding and first 5 minutes experience, which is essential when adopting a new tool.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/scaffold-config/src/dependencies.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export const WIZARD_DEPENDENCY_TYPESCRIPT = {
   package: 'typescript',
   installer: 'typescript',
   description: 'TypeScript is a language for application-scale JavaScript',
-  minVersion: '^=3.0.0 || ^=4.0.0',
+  minVersion: '^=3.0.0 || ^=4.0.0' || '^=5.0.0',
```
Comment (mike-plummer) [inline packages/scaffold-config/src/dependencies.ts]: There's a disconnect between [the docs](https://docs.cypress.io/guides/tooling/typescript-support#Install-TypeScript) and our min TS version here - should we update to 3.4.0 while we're in here?

**Review by mike-plummer — APPROVED** [review]
Nice! Noticed one thing with the existing version range on TS but not strictly related to this issue - feel free to adjudicate

File: packages/scaffold-config/src/dependencies.ts (hunk 1 above)
Comment (lmiller1990) [reply packages/scaffold-config/src/dependencies.ts]: I think the docs 3.4 limitation is specific to E2E - maybe something to do with the usage of `ts-loader`... I have no idea why we need 3.4 or if it's applicable here.

3.4: https://devblogs.microsoft.com/typescript/announcing-typescript-3-4/

**Review by warrensplayer — COMMENTED** [review]
I think the update to support TS 5 is fine, but I have a question about the tests.

@lmiller1990 Why did you replace the TS 4.5 test with a TS 5 test?  It seems like the original TS 4.5 test was there to test that the `preserveValueImports` parameter that was introduced in TS 4.5 does not break earlier versions.  It was introduced here: https://github.com/cypress-io/cypress/pull/20889

Shouldn't we leave this test and just add a new system test for TS 5?

**Comment by lmiller1990** [comment]
@warrensplayer I might have misunderstood the original test, I thought the idea was "make sure it works with TS 4.5+". I was just trying to minimize the number of projects - this should still capture the original intent... but I can make a separate project, that might be more maintainable long term. What do you think?

**Comment by warrensplayer** [comment]
@lmiller1990 Just commenting my Slack conversation here:

> I see your point about not wanting more fixtures.
The original test for 4.5 was released with the code [here](https://github.com/cypress-io/cypress/pull/20889/files#diff-00f0889e22dd0e33307350843f105d66ae5453aa0ea6a4564eb13959c4452e86R26-R29) which specifically looks for TS 4.5 or greater.  My thought at the time was that someone could change that code to say TS 5 or greater and the tests would pass, but TS 4.5 - < TS 5 would be broken.

Not going to block on this.  I will approve.

**Review by warrensplayer — APPROVED** [review]

**Comment by lmiller1990** [comment]
@warrensplayer made a separate test project.
@mike-plummer I pushed a commit clarifying TS support at `^=3.4.0` to be the same as E2E.

**Pull request merged**